### PR TITLE
fixing release workflows

### PR DESCRIPTION
**Pull request by thirtytwobits**
Testing this by manually fixing the toolshed workflow (https://github.com/OpenCyphal/docker_toolchains/actions/runs/4166778533/jobs/7211540974).

### CODE REVIEW

File: .github/workflows/texer.yml [hunk 1]
```diff
@@ -21,11 +21,16 @@ jobs:
...
     - uses: actions/checkout@v3
 
+    - name: process event
+      id: process_event
+      uses: olegtarasov/get-tag@v2.1.2
+      with:
+        tagRegex: "(${{ env.IMAGE_VERSION_PREFIX }}).*"
     - name: dryrun-build
-      if: ${{ github.event_name == 'pull_request' }}
+      if: ${{ github.event_name == 'pull_request' || ( github.event_name == 'release' && startsWith( steps.process_event.outputs.tag, env.IMAGE_VERSION_PREFIX ) && github.event.release.prerelease ) }}
```
Comment (emrainey) [inline .github/workflows/texer.yml:30]: What's the difference here between a prerelease and the pull request?

File: .github/workflows/toxic.yml [hunk 2]
```diff
@@ -21,11 +21,16 @@ jobs:
...
     - uses: actions/checkout@v3
 
+    - name: process event
+      id: process_event
+      uses: olegtarasov/get-tag@v2.1.2
+      with:
+        tagRegex: "(${{ env.IMAGE_VERSION_PREFIX }}).*"
     - name: dryrun-build
-      if: ${{ github.event_name == 'pull_request' }}
+      if: ${{ github.event_name == 'pull_request' || ( github.event_name == 'release' && startsWith( steps.process_event.outputs.tag, env.IMAGE_VERSION_PREFIX ) && github.event.release.prerelease ) }}
```
Comment (emrainey) [inline .github/workflows/toxic.yml:30]: same question here, what's the difference between a pre-release and the pull_request?

File: .github/workflows/texer.yml (hunk 1 above)
Comment (thirtytwobits) [reply .github/workflows/texer.yml:30]: releases can only be initiated, manually, by an administrator. If that actor marks a release as a "pre-release" then the source (read: Dockerfile and associated shell scripts) is archived, the containers are built, but the built containers are not pushed to ghcr. I'm not sure why we'd ever use this but I also didn't want to push a container from a release event that was marked as such.

File: .github/workflows/toxic.yml (hunk 2 above)
Comment (thirtytwobits) [reply .github/workflows/toxic.yml:30]: See answer above.

**Review by emrainey — APPROVED** [review]

**Pull request merged**